kubera/authentication/rest.go

```diff
@@ -0,0 +1,99 @@
+package rest
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/go-resty/resty/v2"
+)
+
+const (
+	kuberaHost = "KUBERA_HOST" //KUBERA_HOST is Env variable contains value of Kubera's Domain/IP
+	kuberaUser = "KUBERA_USER" // Kubera Username
+	kuberaPass = "KUBERA_PASS" //Kubera password
+)
+
+type Token struct {
+	Token     string `json:"jwt"`
+	AccountID string `json:"accountId"`
+}
+
+type KuberaError struct {
+	Err        error
+	Message    string
+	StatusCode int
+}
+
+type Error struct {
+	Message string `json:"message"`
+}
+
+// Connection contains the details of credentials
+type Connection struct {
+	Host     string
+	UserName string
+	Password string
+}
+
+type TokenFetchingConfig struct {
+	Connection
+}
+
+type TokenFetching struct {
+	Connection
+}
+
+//NewTokenFetcher return the credentials of kubera user ,pass and host
+func NewTokenFetcher() (*TokenFetching, error) {
+	host, ok := os.LookupEnv(kuberaHost)
+	if !ok {
+		return nil, errors.New("KUBERA_HOST environment variable required")
+	}
+	userName, ok := os.LookupEnv(kuberaUser)
+	if !ok {
+		return nil, errors.New("KUBERA_USER environment variable required")
+	}
+	userPass, ok := os.LookupEnv(kuberaPass)
+	if !ok {
+		return nil, errors.New("KUBERA_PASS environment variable required")
+	}
+	return &TokenFetching{
+		Connection{
+			Host:     host,
+			UserName: userName,
+			Password: userPass,
+		},
+	}, nil
+}
+
+//Fetch token
+func (a *Connection) Fetch() (*Token, *KuberaError) {
```

Comment:
return Token, KuberaError

Reply:
if it is not a pointer, i think can't return  `nil` .. ! 😕  
pls give me suggestion how to do @AmitKumarDas .

Reply:
return Token{}, KuberaError